packages/ember-environment/lib/index.js

```diff
@@ -66,8 +67,19 @@ ENV.LOG_STACKTRACE_ON_DEPRECATION = defaultTrue(ENV.LOG_STACKTRACE_ON_DEPRECATIO
 */
 ENV.LOG_VERSION = defaultTrue(ENV.LOG_VERSION);
 
-// default false
-ENV.MODEL_FACTORY_INJECTIONS = defaultFalse(ENV.MODEL_FACTORY_INJECTIONS);
+if (DEBUG) {
+  if ('MODEL_FACTORY_INJECTIONS' in ENV) {
+    deprecate(
+      'Ember.MODEL_FACTORY_INJECTIONS is no longer required',
+      false,
+      {
+        id: 'ember-metal.model_factory_injections',
+        until: '3.0.0',
```

Comment:
Hmm, good question. This isn't _really_ a deprecation per-se, just a handy warning that a setting that used to matter doesn't any more.

We can either swap this with a `warn` (instead of `deprecate`) or leave it as a deprecation and say `until: "2.17.0"`...